src/tsickle.ts

```diff
@@ -476,7 +476,18 @@ abstract class ClosureRewriter extends Rewriter {
         // But it's fine to translate TS "implements Class" into Closure
         // "@extends {Class}" because this is just a type hint.
         const typeChecker = this.typeChecker;
-        const sym = this.mustGetSymbolAtLocation(impl.expression);
+        const sym = this.typeChecker.getSymbolAtLocation(impl.expression);
+        if (!sym) {
+          // It's possible for a class declaration to extend an expression that
+          // is not a symbol, for example when a mixin function is used to build
+          // a base class, as in `declare MyClass extends MyMixin(MyBaseClass)`.
+          //
+          // Handling this correctly is tricky. Closure also doesn't handle this
+          // `extends <expression>` syntax. We would probably need to generate
+          // an intermediate class declaration and extend that. For now, just
+          // omit the `extends` annotation.
+          continue;
```

Comment:
Can you emit a comment a la `// taze: cannot resolve extended supertype` to make debugging easier?

Reply:
I added a call to `debugWarn` so ti will show up in the diagnostics. Looks like I could also/instead do `this.emit('* ...')` from inside `maybeAddHeritageClauses`, which would add a comment to the class jsdoc, but I wasn't sure if that was a good idea (what if the jsdoc comment hadn't started yet?). Let me know which you prefer, and if just emitting a string directly from that function is ok.

Reply:
I think it'd be nice to leave a comment in the generated JS. If something goes wrong during compilation, it'll be very helpful to have some documentation of what happened in the code the user sees.

Reply:
Done, PTAL.